src/DocTestCommand.hs

```diff
@@ -0,0 +1,18 @@
+{- $setup
```

Comment:
Making multiple files for doctest stanzas seems unecessary. I suggest using 1 file and use CPP macros to control what lines are shown.
Eg,
```

{- $setup
>>> :set -XFlexibleContexts
>>> :set -XScopedTypeVariables
>>> import Data.Char (toUpper)
>>> import Data.Function ((&))
>>> import qualified Streamly.Console.Stdio as Stdio
>>> import qualified Streamly.Data.Array as Array
>>> import qualified Streamly.Data.Fold as Fold
>>> import qualified Streamly.Data.Stream.Prelude as Stream
>>> import qualified Streamly.System.Process as Process
>>> import qualified Streamly.Unicode.Stream as Unicode
>>> import qualified Streamly.Internal.Console.Stdio as Stdio
>>> import qualified Streamly.Internal.Data.Stream as Stream
>>> import qualified Streamly.Internal.FileSystem.Dir as Dir
#ifdef DOC_PROCESS
>>> import qualified Streamly.Internal.System.Process as Process
#endif
>>> import qualified Streamly.Internal.Unicode.Stream as Unicode
-}
```
and then you can use it like so,
```
#define DOC_PROCESS
#include "DocTestCommon.hs"
```

Reply:
In general, if we use this scheme everywhere, it will become difficult to manage the divergence across modules using conditionals and whenever you change the file for one module you will have to think about all the modules it is included in. So I think it is better to replicate.